app/models/user.rb

```diff
@@ -14,14 +14,15 @@ class User < ActiveRecord::Base
     :twitter_username
   ].freeze
 
+  before_destroy :yank_gems
```

Comment:
I'm okay with that that. Let's make it more clear on the page though. And
maybe include instructions for adding an owner right there too.

On Tuesday, July 12, 2016, Nick Quaranto <notifications@github.com> wrote:

> In app/models/user.rb
> <https://github.com/rubygems/rubygems.org/pull/1301#discussion_r70547314>:
>
> > @@ -14,14 +14,15 @@ class User < ActiveRecord::Base
> >      :twitter_username
> >    ].freeze
> >
> > +  before_destroy :yank_gems
>
> I don't think this should change our policy regarding how gems are dealt
> with. There should always be an owner for a gem. Here's what I think
> deleting your account should do:
>
>    1. For gems with > 1 owner: removes your ownership.
>    2. For gems with 1 owner (just yourself): yanks all versions of that
>    gem.
>
> We can be very explicit and explain this on a special page as you're
> deleting your account. It's not an action to be done mistakenly (but that
> will happen!)
>
> Any objections to this? @dwradcliffe <https://github.com/dwradcliffe>
> @evanphx <https://github.com/evanphx> @indirect
> <https://github.com/indirect>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rubygems/rubygems.org/pull/1301/files/9e00a444a1b185f3d04a63b34113d3171fb786ae#r70547314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAHQQiqQh1C_6ZuJu58v4DAMTyiWjAZdks5qVDJ8gaJpZM4IqAZ2>
> .
>


-- 
---
David Radcliffe